rust/tests/ffi.rs

```diff
@@ -0,0 +1,184 @@
+    assert_eq!(result.status_code, StatusCode::EVMC_SUCCESS);
+}
+
+#[test]
+fn run_no_code() {
+    let mut host = null_ptr_host_interface();
+    host.get_tx_context = Some(get_tx_context_zeroed);
+    let context = ptr::null_mut();
+    let revision = Revision::EVMC_CANCUN;
+    let message = driver::to_evmc_message(&MockExecutionMessage::default());
+    let code = &[];
+    let result = unsafe { driver::run(&host, context, revision, &message, code) };
+    assert_eq!(result.status_code, StatusCode::EVMC_SUCCESS);
+}
+
+#[test]
+fn run_stack_underflow() {
+    let mut host = null_ptr_host_interface();
+    host.get_tx_context = Some(get_tx_context_zeroed);
+    let context = ptr::null_mut();
+    let revision = Revision::EVMC_CANCUN;
+    let message = driver::to_evmc_message(&MockExecutionMessage::default());
+    let code = &[Opcode::Add as u8]; // this will error because the stack is empty
+    let result = unsafe { driver::run(&host, context, revision, &message, code) };
+    assert_eq!(result.status_code, StatusCode::EVMC_STACK_UNDERFLOW);
+}
+
+#[test]
+fn run_steppable_with_mock_context() {
+    let host = host_interface::mocked_host_interface();
+    let mut context = MockExecutionContextTrait::new();
+    context
+        .expect_get_tx_context()
+        .times(1)
+        .return_const(TX_CONTEXT_ZEROED);
+    let revision = Revision::EVMC_CANCUN;
+    let message = driver::to_evmc_message(&MockExecutionMessage::default());
+    let code = &[Opcode::Push0 as u8];
+    let result = unsafe {
+        driver::run_steppable(
+            host,
+            &mut context as *mut MockExecutionContextTrait as *mut ffi::c_void,
+            revision,
+            &message,
+            code,
+            StepStatusCode::EVMC_STEP_RUNNING,
+            0,
+            0,
+            &mut [ZERO],
+            &mut [0],
+            &mut [0],
+            1,
+        )
+    };
+    assert_eq!(result.status_code, StatusCode::EVMC_SUCCESS);
+    assert_eq!(result.step_status_code, StepStatusCode::EVMC_STEP_RUNNING);
+}
+
+#[test]
+fn run_steppable_without_values() {
+    let mut host = null_ptr_host_interface();
+    host.get_tx_context = Some(get_tx_context_zeroed);
+    let context = ptr::null_mut();
+    let revision = Revision::EVMC_CANCUN;
+    let message = driver::to_evmc_message(&MockExecutionMessage::default());
+    let code = &[];
+    let result = unsafe {
+        driver::run_steppable(
+            host,
+            context,
+            revision,
+            &message,
+            code,
+            StepStatusCode::EVMC_STEP_RUNNING,
+            0,
+            0,
+            &mut [],
+            &mut [],
+            &mut [],
+            1,
+        )
+    };
+    assert_eq!(result.status_code, StatusCode::EVMC_SUCCESS);
+    assert_eq!(result.step_status_code, StepStatusCode::EVMC_STEP_STOPPED);
+}
+
+#[test]
+fn run_steppable_with_only_tx_context() {
```

Comment:
What are we testing here? 
please rewrite test name to better reflect property tested:

Reply:
see comment below